client/js/components/content-slider.js

```diff
@@ -97,11 +108,32 @@ const contentSlider = (el, options) => {
     }
   }
 
-  function createWidthArray(slidesArray) {
+  function createWidthArray(slidesArray, containerWidth) {
     const widthArray = [];
-    nodeList(slidesArray).forEach((e, i) => {
-      const elementWidth = e.getBoundingClientRect().width;
-      widthArray.push(elementWidth);
+    const maxWidth = containerWidth;
+    nodeList(slidesArray).forEach((el, i) => {
+      const width = el.offsetWidth;
+      const style = window.getComputedStyle(el);
+      const horizontalMargins = parseInt(style.marginLeft) + parseInt(style.marginRight);
+      const image = el.getElementsByTagName('img')[0];
+      if (image) {
+        image.style.removeProperty('width');
+        image.style.removeProperty('height');
+        const cssHeight = parseInt(window.getComputedStyle(image).getPropertyValue('height'), 10);
+        const imageWidth = image.getAttribute('data-width');
+        const imageHeight = image.getAttribute('data-height');
+        const widthByHeight = imageWidth / imageHeight * cssHeight;
+
+        if (widthByHeight <= maxWidth) {
+          widthArray.push(widthByHeight + horizontalMargins);
+        } else {
+          widthArray.push(maxWidth);
+          image.style.width = `${maxWidth - horizontalMargins}px`;
+          image.style.height = 'auto';
+        }
+      } else {
+        widthArray.push(width + horizontalMargins);
+      }
```

Comment:
There is a lot of mutation in this component. It might be nice to try and get state / side effects decoupled, and then we can wrap the side effects in something like [`fastDOM`](https://github.com/wilsonpage/fastdom)